php/getOrderMessages.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * This example script gets information on Order Messages from the Tradebyte REST API.
+ * Then marks them as exported
+ *
+ * @author Marcos Doellerer<marcos.doellerer@fatchip.de>
+ */
+
+/*
+ * API-Credentials - modify to your personal access credentials
+ *
+ * @see https://tradebyte.io/how-to/generate-rest-api-credentials-in-tb-one/
+ */
+$sApiUser = 'api-user'; // Your API username
+$sApiPassword = 'api-password'; // Your API password
+$sMerchantId = '1234'; // Your digit merchant ID
+$sChannelId = '5678'; // Your digit channel ID
+
+
+/*
+* Get data from REST API with curl
+*/
+$sUrl = "https://rest.trade-server.net/" . $sMerchantId . "/messages/?channel=" . $sChannelId;
+
+$oCurl = curl_init();
+curl_setopt($oCurl, CURLOPT_URL, $sUrl);
+curl_setopt($oCurl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+curl_setopt($oCurl, CURLOPT_USERPWD, $sApiUser . ":" . $sApiPassword);
+curl_setopt($oCurl, CURLOPT_RETURNTRANSFER, 1);
+curl_setopt($oCurl, CURLOPT_HEADER, 0);
+curl_setopt($oCurl, CURLOPT_SSL_VERIFYPEER, false);
+curl_setopt($oCurl, CURLOPT_TIMEOUT, 3600);
+$sResponse = curl_exec($oCurl);
+if ($sResponse === false) {
+    echo 'Error: ' . curl_error($oCurl) . ' ErrorNr: ' . curl_errno($oCurl);
+}
+curl_close($oCurl);
+
+/**
+ * Process data as you need
+ */
+$oXml = simplexml_load_string($sResponse);
+if ($oXml) {
+
+    $messagesImported = [];
+
+    foreach ($oXml->MESSAGE as $oMessage) {
+        $messagesImported[] = (string)$oMessage->TB_ORDER_ID;
+        echo "Message Type: " . (string)$oMessage->MESSAGE_TYPE . PHP_EOL;
+        echo " Order number in Channel : " . (string)$oMessage->CHANNEL_ORDER_ID . PHP_EOL;
+    }
+    /**
+     * Sending confirmation to Tradebyte
+     */
+    sendReceivedFlagToTradebyte($messagesImported, $sMerchantId, $sChannelId, $sApiUser, $sApiPassword);
+
+} else {
+    print_r($sResponse);
+}
+
+function sendReceivedFlagToTradebyte($arrayOfMessageIds, $sMerchantId, $sChannelId, $sApiUser, $sApiPassword){
+    foreach ($arrayOfMessageIds as $messageId){
+
+        $sUrl = "https://rest.trade-server.net/" . $sMerchantId . "/messages/" . $messageId . "/?channel=" . $sChannelId;
```

Comment:
https://rest.trade-server.net/" . $sMerchantId . "/messages/" . $messageId . "/exported

Reply:
Hi Dominik,
what you write does not match the official API documentation - is this a hidden feature or do we use an old version of the docs?
POST https://rest.trade-server.net/(HNR)/messages/(MSGID)/processed?channel=(CHID)
(see https://tradebyte.io/documentation/basics-about-data-exchange/rest-calls-for-channels-and-merchants/)
Thx for your feedback!
Hendrik

Reply:
@Bugfighter accordingly to this link you missing "/processed" in the URL.

Reply:
Correct - so we use "processed?channel= . $sChannelId" and not "/exported".

Reply:
Ok. Correction is in place.